homeassistant/components/bluecurrent/config_flow.py

```diff
@@ -0,0 +1,99 @@
+"""Config flow for BlueCurrent integration."""
+from __future__ import annotations
+
+from collections.abc import Mapping
+from typing import Any
+
+from bluecurrent_api import Client
+from bluecurrent_api.exceptions import (
+    AlreadyConnected,
+    InvalidApiToken,
+    RequestLimitReached,
+    WebsocketException,
+)
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_API_TOKEN
+from homeassistant.data_entry_flow import FlowResult
+
+from .const import DOMAIN, LOGGER
+
+DATA_SCHEMA = vol.Schema({vol.Required(CONF_API_TOKEN): str})
+
+
+async def validate_input(client: Client, api_token: str) -> None:
+    """Validate the user input allows us to connect."""
+    await client.validate_api_token(api_token)
+
+
+async def get_email(client: Client) -> str:
+    """Validate the user input allows us to connect."""
+    email: str = await client.get_email()
+    return email
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle the config flow for Blue Current."""
+
+    VERSION = 1
+
+    input: dict[str, Any] = {}
+    client = Client()
+    entry: config_entries.ConfigEntry | None
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+
+        errors = {}
+        if user_input is not None:
+
+            api_token = user_input[CONF_API_TOKEN]
+            self._async_abort_entries_match({CONF_API_TOKEN: api_token})
+
+            try:
+                await validate_input(self.client, api_token)
```

Comment:
`validate_input` contains 1 call... why not do that call here directly instead? It removes the need for an additional method (which is just playing a proxy).

Reply:
removed validate_input